bundles/org.openhab.binding.mqtt.generic/src/main/java/org/openhab/binding/mqtt/generic/AbstractMQTTThingHandler.java

```diff
@@ -237,11 +273,17 @@ public void dispose() {
 
     @Override
     public void updateChannelState(ChannelUID channelUID, State value) {
+        if (!messageReceived.getAndSet(true)) {
```

Comment:
updateChannelState is called frequently (hot path). This statement will access an atomic variable, which requires cpu cores to ensure cache coherence (expensive!). Is it absolutely necessary to access this variable on each channel state update?

Reply:
Well, we need to keep track, if messages were received (at least for some HA components) to be able to set the thing status.
Would it be better to schedule this check in another thread?

Reply:
At the moment every mqtt Thing or to be more precise every mqtt thing channel pays this cost though and it is not even used at all times, that's my issue with this code.

Scheduling a thread here would be even more costly. I don't have an idea yet.